packages/web/utils/helper.js

```diff
@@ -240,3 +240,38 @@ export const dateToString = (date) => {
 	const [year, month, day] = date.split('-');
 	return `${day.substring(0, 2)}/${month}/${year}`;
 };
+
+/**
+ * Calculates the average rating based on reviews array.
+ *
+ * @param {Array} reviews The array of reviews
+ * @returns {number}
+ */
+export const calculateRatingsAverage = (reviews) => {
+	if (!reviews.length) return null;
+
+	// Format rating count -> { '3': 1, '4': 2 }
+	const ratings = reviews
+		.map((review) => review.rating)
+		.reduce((acc, curr) => {
+			if (typeof acc[curr] === 'undefined') {
+				acc[curr] = 1;
+			} else {
+				acc[curr] += 1;
+			}
+
+			return acc;
+		}, {});
```

Comment:
We usually use `reduce` to replace a situation where we need `filter` + `map`, so, in this case, we can avoid looping through the array twice if we do something like (didn't test the code to make sure it works):
```suggestion
		.reduce((acc, curr) => {
			if (typeof acc[curr.rating] === 'undefined') {
				acc[curr.rating] = 1;
			} else {
				acc[curr.rating] += 1;
			}

			return acc;
		}, {});
```
I'd probably refactor this as well, since `undefined` is a [falsy value](https://developer.mozilla.org/en-US/docs/Glossary/Falsy).
```
const ratings = reviews
		.reduce((acc, curr) => {
    		acc[curr.rating] = acc[cur.rating] ? acc[curr.ratng]+1 : 1;
			return acc;
		},{});
```